src/jit/codegenlegacy.cpp

```diff
@@ -11364,8 +11364,6 @@ void CodeGen::genCodeForTreeSmpOpAsg(GenTreePtr tree)
 
             if (VarSetOps::IsMember(compiler, genUpdateLiveSetForward(op2), varDsc->lvVarIndex))
             {
-                noway_assert(compiler->opts.compDbgCode);
```

Comment:
The assert is saying that, unless this is debug code, we should never encounter a definition of a lclVar that is currently live (i.e. the definition should have killed any uses that are upwards-exposed at this point). So, if this assert fires there is either some valid reason we are keeping it alive, OR there is a liveness bug.